metrics-operator/api/v1alpha3/analysis_types.go

```diff
@@ -0,0 +1,72 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha3
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// AnalysisSpec defines the desired state of Analysis
+type AnalysisSpec struct {
+	//Timeframe specifies the range for the corresponding query in the AnalysisValueTemplate
+	Timeframe `json:"timeframe"`
+	// Args corresponds to a map of key/value pairs that can be used to substitute placeholders in the AnalysisValueTemplate query. The placeholder must be the capitalized version of the key; i.e. for args foo:bar the query could be "query:percentile(95)?scope=tag(my_foo_label:{{.Foo}})".
+	Args map[string]string `json:"args,omitempty"`
+	// AnalysisDefinition refers to the AnalysisDefinition, a CRD that stores the AnalysisValuesTemplates
+	AnalysisDefinition AnalysisDefinitionRef `json:"analysisDefinition"`
+}
+
+// AnalysisDefinitionRef represents the AnalysisDefinition object
+type AnalysisDefinitionRef struct {
+	// Name of the AnalysisDefinition
+	Name string `json:"name"`
+}
+
+//+kubebuilder:object:root=true
+//+kubebuilder:subresource:status
```

Comment:
Let's also add an AnalysisDefinition name to the columns shown when executing `kubectl get`. Something like this

// +kubebuilder:printcolumn:name="AnalysisDefinition",type=string,JSONPath=`.spec.analysisDefinition.name`